### PR TITLE
addloading

### PR DESCRIPTION
**Pull request by Zstro**

### CODE REVIEW

File: src/S3ClientGUI/s3treemodel.cpp [hunk 1]
```diff
@@ -20,6 +20,23 @@
 
 #include "s3treemodel.h"
 
+namespace
```
Comment (thesues) [inline src/S3ClientGUI/s3treemodel.cpp:23]: What's the usage of "namespace" ?

File: src/S3ClientGUI/s3treemodel.cpp [hunk 2]
```diff
@@ -20,6 +20,23 @@
...
+{
+	static void advanceSpinner(QVariantList& loadData)
```
Comment (thesues) [inline src/S3ClientGUI/s3treemodel.cpp:25]: A simple QString could do this, why use QVariantList ?

File: src/S3ClientGUI/s3treemodel.cpp [hunk 3]
```diff
@@ -106,6 +123,28 @@ void S3TreeModel::setRootPath(const QString &path) {
...
     m_currentData.clear();
 
+	//loading
+	QVariantList loadData;
+	loadData << "...loading [-]" << QVariant() << QVariant() << QVariant();
+	m_currentData.append(new SimpleItem(loadData, S3LoadingType, "", ""));
+
+	QObject::connect(timer, &QTimer::timeout, this, [=]()
+	{
+		if (m_currentData.size() > 0)
```
Comment (thesues) [inline src/S3ClientGUI/s3treemodel.cpp:133]: from line 133 ~ 139,  what for ?

File: src/S3ClientGUI/s3treemodel.cpp [hunk 4]
```diff
@@ -143,10 +182,10 @@ void S3TreeModel::setRootPath(const QString &path) {
 
     qDebug() << "prefix name" << prefix;
 
-	//in begining, insert ".." Folder
-	QVariantList data;
+	//in begining, insert ".." Folder---move to listObjectFinished()
```
Comment (thesues) [inline src/S3ClientGUI/s3treemodel.cpp:185]: If you want to remove some code, just REMOVE it ,  do not comment  the code

Comment (Zstro) [reply src/S3ClientGUI/s3treemodel.cpp:185]: I konw

File: src/S3ClientGUI/s3treemodel.cpp (hunk 3 above)
Comment (Zstro) [reply src/S3ClientGUI/s3treemodel.cpp:133]: for just out "loading". there ware out  ".." and "loading" at the same time

File: src/S3ClientGUI/s3treemodel.cpp (hunk 2 above)
Comment (Zstro) [reply src/S3ClientGUI/s3treemodel.cpp:25]: Because I was used QString ,it was had wrong

File: src/S3ClientGUI/s3treemodel.cpp (hunk 1 above)
Comment (Zstro) [reply src/S3ClientGUI/s3treemodel.cpp:23]: I delete it

File: src/S3ClientGUI/s3treemodel.cpp (hunk 3 above)
Comment (Zstro) [reply src/S3ClientGUI/s3treemodel.cpp:133]: I try again,it's ok to delete.

**Pull request closed without merging**